src/support/rand.rs

```diff
@@ -12,15 +12,62 @@ use rand::{
 };
 
 impl<const BITS: usize, const LIMBS: usize> Distribution<Uint<BITS, LIMBS>> for Standard {
+    #[inline]
     fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> Uint<BITS, LIMBS> {
-        let mut limbs = [0; LIMBS];
-        if let Some((last, rest)) = limbs.split_last_mut() {
-            for limb in rest {
-                *limb = rng.gen();
+        <Uint<BITS, LIMBS>>::random_with(rng)
+    }
+}
+
+impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
+    /// Creates a new [`Uint`] with the default cryptographic random number
+    /// generator.
+    ///
+    /// This is currently [`rand::thread_rng`].
+    #[inline]
+    #[must_use]
+    #[cfg(feature = "std")]
+    pub fn random() -> Self {
+        // SAFETY: `uint` is only accessible after random initialization.
+        #[allow(clippy::uninit_assumed_init)]
+        let mut uint = unsafe { core::mem::MaybeUninit::<Self>::uninit().assume_init() };
+        uint.randomize();
+        uint
+    }
+
+    /// Creates a new [`Uint`] with the given random number generator.
+    #[inline]
+    #[doc(alias = "random_using")]
+    #[must_use]
+    pub fn random_with<R: rand::RngCore + ?Sized>(rng: &mut R) -> Self {
+        // SAFETY: `uint` is only accessible after random initialization.
+        #[allow(clippy::uninit_assumed_init)]
+        let mut uint = unsafe { core::mem::MaybeUninit::<Self>::uninit().assume_init() };
+        uint.randomize_with(rng);
+        uint
+    }
+
+    /// Fills this [`Uint`] with the default cryptographic random number
+    /// generator.
+    ///
+    /// See [`random`](Self::random) for more details.
+    #[inline]
+    #[cfg(feature = "std")]
+    pub fn randomize(&mut self) {
+        self.randomize_with(&mut rand::thread_rng());
+    }
+
+    /// Fills this [`Uint`] with the given random number generator.
+    #[inline]
+    #[doc(alias = "randomize_using")]
+    pub fn randomize_with<R: rand::RngCore + ?Sized>(&mut self, rng: &mut R) {
+        if Self::SHOULD_MASK {
```

Comment:
why is this check necessary? if the u64 randomization algo is uniform, then every bit has equal probability of being set and unset, so masking . so can't you simply fill the full width and always mask?

Reply:
yeah i guess that works too